### PR TITLE
update compile to use shards build

### PR DESCRIPTION
**Pull request by robacarp**
Shards now has a `shards build` command which will build all binary outputs of the project. Rather than parsing over the shard.yml and manually selecting a build target, this simply builds with shards.

Thoughts?

### CODE REVIEW

**Comment by jwoertink** [comment]
It looks like there's this line that builds the binary to a file named `app`. When you run `shards build`, doesn't that use the name listed in `targets`? (your app name). I think we'd have to change [this line](https://github.com/luckyframework/lucky_cli/blob/master/src/web_app_skeleton/Procfile#L1) because the binary would no longer be named `app`, but instead whatever your app is called... Unless you know of a way to tell `shards build` to call it "app"?

**Comment by robacarp** [comment]
Yeah, I'd buy that. Here's what I have in my files:

## shard.yml
```yml
targets:
  server:
    main: src/start_server.cr
  worker:
    main: src/app_worker.cr
```

## procfile
```procfile
web: ./bin/server
worker: ./bin/worker
release: lucky db.migrate
```

If this is a good direction to go in, I can make tweaks to the lucky_cli repo to generate these files as such.

**Comment by jwoertink** [comment]
I think I'm good with the change, we just need to make sure we account for this working on a fresh app setup. This will be a breaking change for current apps like [the website](https://github.com/luckyframework/website/blob/master/Procfile#L1). 

If we can confirm that this is always the case, then I'll merge it in:

```yml
targets:
  super_app_8:  # <--- whatever this is 
     main: src/super_app_8.cr
```

```
web: ./bin/super_app_8  # <--- is always the same here?
```

**Comment by robacarp** [comment]
This is a complicated breaking change for sure. I don't know what the precedent for making breaking changes to a buildpack is.

The shard.yml file [isn't currently provided by the lucky cli](https://github.com/luckyframework/lucky_cli/blob/356b8fa53b8434bbbf7ba95e758c96b0abde15d9/src/generators/web.cr#L143-L149) it's provided by `crystal init`

My local crystal is 0.35, but this is what it generates with a lucky init:

```diff
~/d/repositories> lucky init testproj3
Welcome to Lucky v0.24.0 🎉

+Project name?: testproj3
-[snip]

~/d/repositories> cat testproj3/shard.yml
+name: testproj3
version: 0.1.0

targets:
+  testproj3:
    main: src/testproj3.cr

crystal: 0.35.1

/d/repositories> cat testproj3/Procfile
+web: ./app
release: lucky db.migrate
```

Procfile.dev runs `lucky watch` which builds and runs `start_server.cr` which is an identical entrypoint to `{shard_name}.cr`

**Review by jwoertink — APPROVED** [review]
Ok, I think we just need to update LuckyCli so the generated Procfile calls the name of the app from the bin directory. (e.g. `web: ./bin/testproj3`)

**Comment by robacarp** [comment]
@jwoertink I submitted pulls on lucky_cli and lucky/website which should update both for this change.

File: bin/compile [hunk 1]
```diff
@@ -77,16 +77,9 @@ start "Installing Dependencies"
...
-    crystal build app.cr --release $NO_DEBUG_IF_AVAILABLE
-    finished
-else
-    eval $(parse_yaml shard.yml "shard_")
-    start "Compiling src/${shard_name}.cr (auto-detected from shard.yml)"
-    crystal build src/${shard_name}.cr --release -o app
-    finished
-fi
+start "Running shards builds --release"
+shards build --release
```
Comment (jwoertink) [inline bin/compile]: Oh, does this need the `--production` flag? `shards build --release --production`. I think that is so it doesnt' build ameba targets and stuff like that...

**Review by jwoertink — APPROVED** [review]
Let's give this a shot!

**Pull request merged**